tests/rules/no-dupe-class-members.js

```diff
@@ -0,0 +1,218 @@
+    "class A { 1() {} 2() {} }",
+    "class A { foo = 1; bar = 1; }",
+    "class A { static foo = 1; foo = 1; }",
+    "class A { foo = 1; } class B { foo = 1; }",
+    "class A { [foo] = 1; foo = 1; }",
+    "class A { foo; bar = 1; }",
+    "class A { foo = () => {}; bar = () => {}; }",
+    "class A { foo = this.foo.bind(this); foo() {} }",
+  ],
+  invalid: [
+    {
+      code: "class A { foo() {} foo() {} }",
+      errors: [
+        {
+          type: "MethodDefinition",
+          line: 1,
+          column: 20,
+          messageId: "unexpected",
+          data: { name: "foo" }
+        }
+      ]
+    },
+    {
+      code: "!class A { foo() {} foo() {} };",
+      errors: [
+        {
+          type: "MethodDefinition",
+          line: 1,
+          column: 21,
+          messageId: "unexpected",
+          data: { name: "foo" }
+        }
+      ]
+    },
+    {
+      code: "class A { 'foo'() {} 'foo'() {} }",
+      errors: [
+        {
+          type: "MethodDefinition",
+          line: 1,
+          column: 22,
+          messageId: "unexpected",
+          data: { name: "foo" }
+        }
+      ]
+    },
+    {
+      code: "class A { 10() {} 1e1() {} }",
+      errors: [
+        {
+          type: "MethodDefinition",
+          line: 1,
+          column: 19,
+          messageId: "unexpected",
+          data: { name: "10" }
+        }
+      ]
+    },
+    {
+      code: "class A { foo() {} foo() {} foo() {} }",
+      errors: [
+        {
+          type: "MethodDefinition",
+          line: 1,
+          column: 20,
+          messageId: "unexpected",
+          data: { name: "foo" }
+        },
+        {
+          type: "MethodDefinition",
+          line: 1,
+          column: 29,
+          messageId: "unexpected",
+          data: { name: "foo" }
+        }
+      ]
+    },
+    {
+      code: "class A { static foo() {} static foo() {} }",
+      errors: [
+        {
+          type: "MethodDefinition",
+          line: 1,
+          column: 27,
+          messageId: "unexpected",
+          data: { name: "foo" }
+        }
+      ]
+    },
+    {
+      code: "class A { foo() {} get foo() {} }",
+      errors: [
+        {
+          type: "MethodDefinition",
+          line: 1,
+          column: 20,
+          messageId: "unexpected",
+          data: { name: "foo" }
+        }
+      ]
+    },
+    {
+      code: "class A { set foo(value) {} foo() {} }",
+      errors: [
+        {
+          type: "MethodDefinition",
+          line: 1,
+          column: 29,
+          messageId: "unexpected",
+          data: { name: "foo" }
+        }
+      ]
+    },
+    {
+      code: "class A { foo = 1; foo = 1; }",
+      errors: [
+        {
+          type: "ClassProperty",
+          line: 1,
+          column: 20,
+          messageId: "unexpected",
+          data: { name: "foo" }
+        }
+      ]
+    },
+    {
+      code: "class A { foo = 1; foo = 1; foo = 1; }",
```

Comment:
```suggestion
      code: "class A { foo = 1; foo = 2; foo = 3; }",
```
so we make sure to check different values too